tsl/test/sql/correlated_constraints.sql

```diff
@@ -0,0 +1,161 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+\set PREFIX 'EXPLAIN (costs off, timing off, summary off)'
+
+CREATE OR REPLACE VIEW compressed_chunk_info_view AS
+SELECT
+   h.schema_name AS hypertable_schema,
+   h.table_name AS hypertable_name,
+   c.schema_name || '.' || c.table_name as chunk_name,
+   c.status as chunk_status
+FROM
+   _timescaledb_catalog.hypertable h JOIN
+  _timescaledb_catalog.chunk c ON h.id = c.hypertable_id
+   LEFT JOIN _timescaledb_catalog.chunk comp
+ON comp.id = c.compressed_chunk_id
+;
+
+CREATE TABLE sample_table (
+       time TIMESTAMP WITH TIME ZONE NOT NULL,
+       sensor_id INTEGER NOT NULL,
+       cpu double precision null,
+       temperature double precision null,
+       name varchar(100) default 'this is a default string value'
+);
+
+SELECT * FROM create_hypertable('sample_table', 'time',
+       chunk_time_interval => INTERVAL '2 months');
+
+\set start_date '2022-01-28 01:09:53.583252+05:30'
+
+INSERT INTO sample_table
+    SELECT
+       	time + (INTERVAL '1 minute' * random()) AS time,
+       		sensor_id,
+       		random() AS cpu,
+       		random()* 100 AS temperature
+       	FROM
+       		generate_series(:'start_date'::timestamptz - INTERVAL '1 months',
+                            :'start_date'::timestamptz - INTERVAL '1 week',
+                            INTERVAL '1 hour') AS g1(time),
+       		generate_series(1, 8, 1 ) AS g2(sensor_id)
+       	ORDER BY
+       		time;
+
+\set start_date '2023-03-17 17:51:11.322998+05:30'
+
+-- insert into new chunks
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 12, 21.98, 33.123, 'new row1');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 12, 17.66, 13.875, 'new row1');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 13, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 9, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 14, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 15, 0.988, 33.123, 'new row3');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 16, 4.6554, 47, 'new row3');
+
+-- Non-int, date, timestamp cannot be specified as a correlated constraint for now
+-- We could expand to FLOATs, NUMERICs later
+\set ON_ERROR_STOP 0
+SELECT * FROM add_dimension('sample_table', 'name', correlated => true);
+SELECT * FROM add_dimension('sample_table', by_correlation('name'));
+\set ON_ERROR_STOP 1
```

Comment:
I think we should only use the new interface for this:

```suggestion
-- Non-int, date, timestamp cannot be specified as a correlated constraint for now
-- We could expand to FLOATs, NUMERICs later
\set ON_ERROR_STOP 0
SELECT * FROM add_dimension('sample_table', by_correlation('name'));
\set ON_ERROR_STOP 1
```

Reply:
Ok, done.